src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -147,6 +147,11 @@ private void InterceptCallAndAdjustArguments(
                 return;
             }
 
+            if (interceptor.GetIsNewExtensionMember() && interceptor.TryGetCorrespondingExtensionImplementationMethod() is { } implementationMethod)
```

Comment:
>interceptor.TryGetCorrespondingExtensionImplementationMethod() is { } implementationMethod [](http://example.com/codeflow?start=57&length=90)

It doesn't feel like we can continue otherwise. If there is a guarantee that the implementation method is present (for example, it is always defined in the same source module), then we should throw unreachable. If there is no guarantee, then we should rep[ort an error and recover.